src/core/client/stream/test/comments/components/CommentContainer.spec.tsx

```diff
@@ -0,0 +1,212 @@
+      Query: {
+        story: () => commentsWithNoUsernames,
+        stream: () => commentsWithNoUsernames,
+      },
+    },
+  });
+
+  const firstComment = commentsWithNoUsernames.comments.edges[0].node;
+  const firstCommentAuthor = stories[0].comments.edges[0].node.author;
+  const firstCommentElement = await within(container).findByTestId(
+    `comment-${firstComment.id}`
+  );
+  expect(
+    within(firstCommentElement).queryByText(firstCommentAuthor!.username!)
+  ).toBeNull();
+
+  expect(
+    within(firstCommentElement).getByText(firstComment.body!)
+  ).toBeDefined();
+});
+
+it("renders InReplyTo", async () => {
+  const { container } = await createTestRenderer({
+    resolvers: {
+      Query: {
+        story: () => storyWithReplies,
+        stream: () => storyWithReplies,
+      },
+    },
+  });
+
+  const firstComment = storyWithReplies.comments.edges[1].node;
+  const firstReply = firstComment.replies.edges[0].node;
+
+  const firstReplyElement = await within(container).findByTestId(
+    `comment-${firstReply.id}`
+  );
+
+  const inReplyTo = within(firstReplyElement).getByText((content, element) => {
+    const match =
+      content.startsWith("In reply") &&
+      !!element &&
+      element.innerHTML.includes(firstComment.author!.username!);
+    return match;
+  });
+
+  expect(inReplyTo).toBeDefined();
+});
+
+it("renders disabled reply when commenting has been disabled site-wide", async () => {
+  const disabledComentingSettings = {
+    ...settings,
+    disableCommenting: {
+      enabled: true,
+      message: "Commenting has been disabled",
+    },
+  };
+  const { container } = await createTestRenderer({
+    resolvers: {
+      Query: {
+        settings: () => disabledComentingSettings,
+      },
+    },
+  });
+
+  const firstComment = stories[0].comments.edges[0].node;
+  const firstCommentElement = within(container).getByTestId(
+    `comment-${firstComment.id}`
+  );
+
+  expect(firstCommentElement).toBeInTheDocument();
+  const replyButton = within(firstCommentElement).getByLabelText("Reply", {
+    exact: false,
+  });
+  expect(replyButton).toBeInTheDocument();
+  expect(replyButton).toBeDisabled();
+});
+
+it("renders disabled reply when story is closed", async () => {
+  const closedStory = {
+    ...stories[0],
+    isClosed: true,
+  };
+
+  const { container } = await createTestRenderer({
+    resolvers: {
+      Query: {
+        story: () => closedStory,
+        stream: () => closedStory,
+      },
+    },
+  });
+
+  const firstComment = stories[0].comments.edges[0].node;
+  const firstCommentElement = within(container).getByTestId(
+    `comment-${firstComment.id}`
+  );
+
+  expect(firstCommentElement).toBeInTheDocument();
+  const replyButton = within(firstCommentElement).getByLabelText("Reply", {
+    exact: false,
+  });
+  expect(replyButton).toBeInTheDocument();
+  expect(replyButton).toBeDisabled();
+});
+
+it.only("renders with tombstone when comment has been deleted", async () => {
```

Comment:
Please remove this to make sure it's running all the tests :)

Reply:
✅